src/filesplugin.js

```diff
@@ -29,160 +50,155 @@ function openChannelSelector(files) {
-			axios.get(urlCheckConnection).then((response) => {
-				const fileIdsStr = response.data.file_ids_to_send_after_oauth
-				const currentDir = response.data.current_dir_after_oauth
-				if (fileIdsStr && currentDir) {
-					this.sendFileIdsAfterOAuth(fileList, fileIdsStr, currentDir)
-				} else {
-					if (DEBUG) console.debug('[Mattermost] nothing to send')
+		const fileIds = fileIdsStr.split(',')
+		const files = fileIds.map((fid) => {
+			const f = nodes.find((n) => n.fileid === parseInt(fid))
+			if (f) {
+				return {
+					id: f.fileid,
+					name: f.basename,
+					type: f.type,
+					size: f.size,
 				}
-			}).catch((error) => {
-				console.error(error)
-			})
-		},
-
-		/**
-		 * In case we successfully connected with oauth and got redirected back to files
-		 * actually go on with the files that were previously selected
-		 *
-		 * @param {object} fileList the one from attach()
-		 * @param {string} fileIdsStr list of files to send
-		 * @param {string} currentDir path to the current dir
-		 */
-		sendFileIdsAfterOAuth: (fileList, fileIdsStr, currentDir) => {
-			if (DEBUG) console.debug('[Mattermost] in sendFileIdsAfterOAuth, fileIdsStr, currentDir', fileIdsStr, currentDir)
-			// this is only true after an OAuth connection initated from a file action
-			if (fileIdsStr) {
-				// trick to make sure the file list is loaded (didn't find an event or a good alternative)
-				// force=true to make sure we get a promise
-				fileList.changeDirectory(currentDir, true, true).then(() => {
-					const fileIds = fileIdsStr.split(',')
-					const files = fileIds.map((fid) => {
-						const f = fileList.files.find((e) => e.id === parseInt(fid))
-						if (f) {
-							return {
-								id: f.id,
-								name: f.name,
-								type: f.type,
-								size: f.size,
-							}
-						}
-						return null
-					}).filter((e) => e !== null)
-					if (DEBUG) console.debug('[Mattermost] in sendFileIdsAfterOAuth, after changeDirectory, files:', files)
-					if (files.length) {
-						if (DEBUG) console.debug('[Mattermost] in sendFileIdsAfterOAuth, after changeDirectory, call openChannelSelector')
-						openChannelSelector(files)
-					}
-				})
 			}
-		},
-
-		connectToMattermost: (selectedFiles = []) => {
-			oauthConnectConfirmDialog(OCA.Mattermost.mattermostUrl).then((result) => {
-				if (result) {
-					if (OCA.Mattermost.usePopup) {
-						oauthConnect(OCA.Mattermost.mattermostUrl, OCA.Mattermost.clientId, null, true)
-							.then((data) => {
-								OCA.Mattermost.mattermostConnected = true
-								openChannelSelector(selectedFiles)
-							})
-					} else {
-						const selectedFilesIds = selectedFiles.map(f => f.id)
-						oauthConnect(
-							OCA.Mattermost.mattermostUrl,
-							OCA.Mattermost.clientId,
-							'files--' + OCA.Files.App.fileList._currentDirectory + '--' + selectedFilesIds.join(','),
-						)
-					}
-				}
-			})
-		},
+			return null
+		}).filter((e) => e !== null)
```

Comment:
`e != null`

Reply:
Hmmm is it a problem to use `!==` when it's not necessary?

Reply:
Not a problem for this one but generally should be used like that since `e !== null` evaluates to true with `undefined`